sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -493,6 +493,21 @@ class CatalogSuite
     }
   }
 
+  test("clone SessionCatalog") {
+    // need to test tempTables are cloned
+    assert(spark.catalog.listTables().collect().isEmpty)
+
+    createTempTable("my_temp_table")
+    assert(spark.catalog.listTables().collect().map(_.name).toSet == Set("my_temp_table"))
+
+    val forkedSession = spark.cloneSession()
```

Comment:
This is just testing the user-facing wrappers. There are also tests for the actual implementation which is `SessionCatalog`. This tests `Catalog(CatalogImpl(SessionCatalog))`